File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid):
        count = 0

        for row in grid:
            first_negative = len(row)
            for i, value in enumerate(row):
                if value < 0:
                    first_negative = i
                    break
            count += len(row) - first_negative

        return count

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int i = grid.size() - 1;
        int j = 0;
        int rowSize = grid.at(0).size();
        
        int result = 0;
        
        while (i >= 0 && j < rowSize) {
            if (grid.at(i).at(j) < 0) {
                result += rowSize - j;
                i -= 1;
            } else {
                j += 1;
            }
        }
        
        return result;
    }
};